### ISSUE TITLE
Add option to show wire overlaps/crossovers as jumps

### ISSUE THREAD

**Issue by flotwig** [LibrePCB/LibrePCB#1414]
##### SUMMARY

Currently, wires that overlap/cross over each other without connecting are displayed as line segments. Despite the fact that connections between wires are shown with a circle, it can be difficult to tell at a glance if two wires are connected or not. Example:

![image](https://github.com/user-attachments/assets/2cd1b2b8-3d89-4b15-a8a9-406678e011ef)

##### SOLUTION

It would be nice to have the option to display non-connecting wires that cross over each other as "jumps", with a curved line. This would help to tell at a glance if two wires should connect. Example:

![image](https://github.com/user-attachments/assets/2cbce5db-67a3-44cb-b5d0-9a466cf106af)

**Comment by ubruhin** [comment]
Really not sure how this could be implemented (not as trivial as it sounds) but I totally agree this would be nice :+1:

**Comment by xotho** [comment]
Why not use just DOT on every wire connected together and no DOT for all the overlaps/crossovers?

So all connection including wire to IC's and so on should always have a DOT if connected together....

Exemples:
![image](https://github.com/user-attachments/assets/e5f2fe3c-3ba0-4ba0-b128-b59acb71e1ad)

![image](https://github.com/user-attachments/assets/df1e653a-4a9d-46bf-9281-a0b2ffed2c5b)

![image](https://github.com/user-attachments/assets/f5df14f3-ef7c-4022-8584-018692329610)

take care,
X!

**Comment by flotwig** [comment]
> Why not use just DOT on every wire connected together and no DOT for all the overlaps/crossovers?
> 
> So all connection including wire to IC's and so on should always have a DOT if connected together....

@xotho This seems to be how LibrePCB currently displays connected/unconnected crossovers, but it can be difficult to read at a glance (in my opinion). Please let me know if I misinterpreted what you were saying.

**Comment by ubruhin** [comment]
> Why not use just DOT on every wire connected together and no DOT for all the overlaps/crossovers?
> So all connection including wire to IC's and so on should always have a DOT if connected together....

IMHO that would be very distracting and also does not conform to how schematics (have to) look.

It's already clear anyway which lines are connected or not because LibrePCB automatically adds the dot on junctions with more than 2 legs. The suggestion of @flotwig would just make it even more clear.

**Comment by xotho** [comment]
Yes Flotwig, as it is now it is ok from my private point of view. The only thing that I would consider is to show DOT at device connection level anyway.

**Actual implementation** where A are not shown as DOT....
![image](https://github.com/user-attachments/assets/6d71490d-9609-4c8c-b593-44fa576c7846)

Device connection level with DOT
![image](https://github.com/user-attachments/assets/b9d33bfd-6a72-43fd-af6a-cf46f04d473b)

An then, in the **Actual implementation** I would also change color of the annular connection (B) to show lines that have to be connected or with optional connection .... or other classes still to be defined .....

Personally the "Jumps" are not a good solution due to the effort in development and the dimension .... when zoomming-in and zoomming-out and it should be an element that take off attention when checking (for me), but I understand the request.

Take care,
X!

**Comment by giancarlo64** [comment]
I personally think that the "semicircle" to indicate that there is no connection is only an graphical and aesthetically solution and has nothing to do with a electrical circuit. If you have more than 1 line to cross it looks horrible.
![jump](https://github.com/user-attachments/assets/44b66c5a-5940-4836-9b2d-aa5ce84da6b5)

I think that electrical diagrams made in this way are unwatchable. 

but that's just my opinion

Gian Carlo

**Comment by xotho** [comment]
Why not adding a set-up option to toggle between the two ?

Take care
X!
